Without this patch the compiler won't add `__decltype(import)` and
`__decltype(export)` directive in front of separately compiled (non inlined)
functionsld. Causing issues when building asio as a separate library.

--- a/boost/asio/detail/config.hpp
+++ b/boost/asio/detail/config.hpp
47c47
< # if defined(_MSC_VER) || defined(__BORLANDC__) || defined(__CODEGEARC__)
---
> # if defined(_MSC_VER) || defined(__BORLANDC__) || defined(__CODEGEARC__) || defined(__MINGW32__)
